Classes/Common/Doc.php

```diff
@@ -1190,8 +1190,6 @@ protected function __construct($location, $pid, $preloadedDocument)
         $this->setPreloadedDocument($preloadedDocument);
         $this->init();
         $this->establishRecordId($pid);
-        $this->logger = GeneralUtility::makeInstance(LogManager::class)->getLogger();
-
```

Comment:
The problem with this logger is:
* it should be already available (in Extbase context implmenting the `LoggerAwareInterface`) but this does not work here.
* the `__wakeup` is only called, if the XML comes from Cache (means get's deserialized).
This is for sure not the finaly beauty. But for now...

Reply:
Yes. I wondered if it would still be slightly cleaner to logger instantiation to the outermost position that works.

Reply:
I would prefer this, too. But I didn't find a better place to make it work in all situations... I wouldn't search any further because I hope to find a better solution later on.